apps/backend/src/adoption/repositories/mongodb/adoption-mongo.repository.ts

```diff
@@ -0,0 +1,74 @@
+import { InjectModel } from '@nestjs/mongoose';
+import {
+  AdoptionRepository,
+  FindNearestType,
+  IAdoption,
+} from '../adoption-repository.interface';
+import { AdoptionDocument } from './adoption-mongo.interface';
+import { Model } from 'mongoose';
+import { AdoptionMap } from './adoption-mongo.map';
+
+export const RADIUS_OF_EARTH = 63781; // km
+
+export class AdoptionMongoDBRepository implements AdoptionRepository {
+  constructor(
+    @InjectModel('Adoption')
+    private readonly adoptionModel: Model<AdoptionDocument>,
+  ) {}
+
+  async create(adoption: IAdoption) {
```

Comment:
What happens if the creating process fails due to invalidate data? How do you handle that back to the user?

Reply:
<img width="479" alt="Screen Shot 2023-05-07 at 10 59 28" src="https://user-images.githubusercontent.com/19699724/236682008-f3859a8f-e69e-4858-84c7-225df3555b5f.png">

Nest has a default error validator. In case of missing data, it returns a 400 error.